### PR TITLE
Add an example of running job in debug mode

### PR DESCRIPTION
**Pull request by xyxxxxx**

### CODE REVIEW

File: job/debug-mode/job_debug.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+metadata:
+  name: torch-mnist-trainingjob-debug
+spec:
+  runMode:
+    debug:
+      enable: true
+      replicaSpecs:
+        - type: worker
+          skipInitContainer: true
+  tensorboardSpec:
```
Comment (ppkube) [inline job/debug-mode/job_debug.yaml:12]: remove `spec. tensorboardSpec`

File: job/debug-mode/job_debug.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            runAsUser: 1000
+          containers:
+            - name: pytorch
+              command:
+                - python
+                - job/pytorchtrainingjob/ddp/torch_mnist_trainingjob.py
+                - "--log_dir"
+                - "job/pytorchtrainingjob/ddp/log"
+                - "--backend"
+                - "gloo"
```
Comment (ppkube) [inline job/debug-mode/job_debug.yaml:32]: use a GPU training example as it exposes a common use case

File: job/debug-mode/README.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+```shell
+kubectl get pod -w
+```
+
+然后使用两个终端分别进入这两个 Pod，
+
+```shell
+# in one terminal
+kubectl exec -it torch-mnist-trainingjob-debug-worker-0 -- bash
```
Comment (ppkube) [inline job/debug-mode/README.md:28]: use a GPU example and demonstrate using `nvidia-smi` to inspect available GPUs

File: job/debug-mode/README.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
+# 使用 Job 的 debug 模式进行调试
+
+本示例使用 Job 的 debug 模式这一功能对要运行的计算任务进行调试。
+
+在 debug 模式下，所有 replica 的容器的启动命令都会被替换为 `sleep inf`，这时由用户进入容器执行任意命令以进行调试。用户可以自由地使用各种命令检查环境，或尝试启动训练（这里训练脚本出错不会造成 Job 的失败）。
```
Comment (ppkube) [inline job/debug-mode/README.md:5]: explain debug a bit more, including additional details such as replacing `command` field

Comment (tswangdi) [reply job/debug-mode/README.md:5]:
```yaml
spec:
  runMode:
    debug:
      enable: true
      replicaSpecs:
        - type: worker
          skipInitContainer: true
          command: ["/bin/usr/sshd"]
```

replicaSpecs 中的配置：
1. skipInitContainer：忽略副本定义中的初始化容器，默认为 false
2. command：使用哪种命令启动副本，默认为 sleep inf
3. 如果一个 replica 在 debug replicaSpec 中未设置，则使用上述默认值。如果没有设置 replicaSpec 字段，则所有 replica 都使用默认值

**Review by ppkube — APPROVED** [review]

**Pull request merged**